src/Abp/Threading/CancellationTokenProviderExtensions.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Threading;
+
+namespace Abp.Threading
+{
+    public static class CancellationTokenProviderExtensions
+    {
+        public static CancellationToken FallbackToProvider(this ICancellationTokenProvider provider, CancellationToken prefferedValue = default)
+        {
+            return prefferedValue == default || prefferedValue == CancellationToken.None
+                ? provider.Token
+                : prefferedValue;
```

Comment:
1. I believe it is incorrect to force the cancellation token to `provider.Token` if a caller explicitly passes `CancellationToken.None`, which is also equal to `default`.

   While cancelling is usually desirable in a purely Get API, since there is no longer anyone to use the information, we may still want to finish Create/Update/Delete.
   See [https://stackoverflow.com/questions/50329618/should-i-always-add-cancellationtoken-to-my-controller-actions](https://stackoverflow.com/q/50329618/8601760).

   I suggest passing the `cancellationToken` from the `Controller` and allowing developers to override it, not least to restore the behaviour before this PR.

2. `preffered` -> `preferred`

Reply:
@hikalkan What do you think?

Reply:
1.  I think you have a point. In my opinion, this is more likely business logic decision.

    > While cancelling is usually desirable in a purely Get API, since there is no longer anyone to use the information, we may still want to finish Create/Update/Delete.
  
    Though I witnessed users clicking `save` button and suddenly closing browser out of panic because of incorrect input they entered for their long running create/update/delete business flow. So in this case we may not want to finish Create/Update/Delete.

    Since it may be business related, developers should have control over what operations may be cancelled.

2. I shall fix it in next commit.

Reply:
@acjh thank you for your comment. You are very carefull and thoughtfull as always. And yyou are right. If a user passes `CancellationToken.None` we are overriding it in this case. A solution would be making it nullable so we use the cancellationtokenprovider only if it is null, instead of `CancellationToken.None`. What do you think?

Reply:
Yes, that solves the correctness of the fallback logic and makes the extension unnecessary.

```diff
- return CancellationTokenProvider.FallbackToProvider(preferredValue);
+ return preferredValue ?? CancellationTokenProvider.Token;
```

## Other considerations

> to restore the behaviour before this PR.

It seems possible with:

```c#
Configuration.ReplaceService(typeof(ICancellationTokenProvider), () =>
{
    IocManager.IocContainer.Register(
        Component.For<ICancellationTokenProvider>().Instance(NullCancellationTokenProvider.Instance)
    );
});
```

>> we may still want to finish Create/Update/Delete.
>
> developers should have control over what operations may be cancelled.

Agreed. Not necessarily in this PR, but it seems possible to implement a `Use` scope similar to `IAbpSession.Use`/`IEntityChangeSetReasonProvider.Use`:

```c#
using (CancellationTokenProvider.Use(CancellationToken.None))
{
    // Operations in this scope will not be cancelled
    return await base.CreateAsync(input);
}
```

Reply:
> It seems possible with:
> ```
> Configuration.ReplaceService(typeof(ICancellationTokenProvider), () =>
> {
>     IocManager.IocContainer.Register(
>         Component.For<ICancellationTokenProvider>().Instance(NullCancellationTokenProvider.Instance)
>     );
> });
> ```
I am not exactly sure about this ReplaceService configuration be done. `AbpKernelModule`? `AbpEntityFrameworkCommonModule`? Can you point me to right direction?

Reply:
It's for developers to restore the behaviour before this PR, so it won't be in this repo.
It would probably be in their Core module.